AudioManagerUtilities/src/CAmCommonAPIWrapper.cpp

```diff
@@ -115,151 +143,214 @@ CAmCommonAPIWrapper* CAmCommonAPIWrapper::getInstance()
+void CAmCommonAPIWrapper::deregisterDispatchSource(CommonAPI::DispatchSource* dispatchSource)
 {
-    for (auto dispatchSourceIterator = mRegisteredDispatchSources.begin();
-                            dispatchSourceIterator != mRegisteredDispatchSources.end();
-                            dispatchSourceIterator++)
+    for(IteratorArrayDispatchSources dispatchSourceIterator = mRegisteredDispatchSources.begin(); dispatchSourceIterator != mRegisteredDispatchSources.end(); dispatchSourceIterator++) 
     {
-        int64_t dispatchTimeout(CommonAPI::TIMEOUT_INFINITE);
-        if(dispatchSourceIterator->second->prepare(dispatchTimeout))
+        if( *dispatchSourceIterator == dispatchSource ) 
         {
-            while (dispatchSourceIterator->second->dispatch());
+            mRegisteredDispatchSources.erase(dispatchSourceIterator);
+            break;
         }
     }
 }
 
-void CAmCommonAPIWrapper::registerDispatchSource(CommonAPI::DispatchSource* dispatchSource, const CommonAPI::DispatchPriority dispatchPriority)
+void CAmCommonAPIWrapper::deregisterAllDispatchSource()
 {
-    mRegisteredDispatchSources.insert({dispatchPriority, dispatchSource});
+    mRegisteredDispatchSources.clear();
 }
 
-void CAmCommonAPIWrapper::deregisterDispatchSource(CommonAPI::DispatchSource* dispatchSource)
+void CAmCommonAPIWrapper::registerWatch(CommonAPI::Watch* watch, const CommonAPI::DispatchPriority)
 {
-    for(auto dispatchSourceIterator = mRegisteredDispatchSources.begin();
-            dispatchSourceIterator != mRegisteredDispatchSources.end();
-            dispatchSourceIterator++) {
+    logInfo(__PRETTY_FUNCTION__);
+    pollfd pollfd_ (watch->getAssociatedFileDescriptor());
+    sh_pollHandle_t handle (0);
 
-        if(dispatchSourceIterator->second == dispatchSource) {
-            mRegisteredDispatchSources.erase(dispatchSourceIterator);
-            break;
-        }
+    am_Error_e error = mpSocketHandler->addFDPoll(pollfd_.fd, pollfd_.events, &pCommonPrepareCallback, &pCommonFireCallback, &pCommonCheckCallback, &pCommonDispatchCallback, watch, handle);
+
+    //if everything is alright, add the watch and the handle to our map so we know this relationship
+    if (error != am_Error_e::E_OK || handle == 0)
+    {
+        logError(__func__,"entering watch failed");
     }
+    else
+        mMapWatches.insert(std::make_pair(handle,watch));
 }
 
 void CAmCommonAPIWrapper::deregisterWatch(CommonAPI::Watch* watch)
 {
-    for(std::map<int,CommonAPI::Watch*>::iterator iter(mMapWatches.begin());iter!=mMapWatches.end();iter++)
+    for(IteratorMapWatches iter=mMapWatches.begin();iter!=mMapWatches.end();iter++)
     {
         if (iter->second == watch)
         {
+            mpSocketHandler->removeFDPoll(iter->first);
             mMapWatches.erase(iter);
             break;
         }
     }
 }
 
+void CAmCommonAPIWrapper::deregisterAllWatches()
+{
+    for(IteratorMapWatches iter=mMapWatches.begin();iter!=mMapWatches.end();iter++)
```

Comment:
spaces

Reply:
...and there is a lot more  called _whitespace errors_ in this patch-set.  A simple git diff will mark in red things like unnecessary spaces at the end of lines.  It doesn't look good to people who are picky about these things, and it is visually disturbing when trying to review the code changes.